### PR TITLE
move multicast tests to truth

### PR DESCRIPTION
**Pull request by yigit**
Please see our contributing guidelines (contributing.md) primarily make sure to sign our cla as we cannot accept code externally without a signed cla

https://opensource.dropbox.com/cla/

### CODE REVIEW

**Comment by yigit** [comment]
issue ##21

File: multicast/src/test/kotlin/com/dropbox/flow/multicast/ChannelManagerTest.kt [hunk 1]
```diff
@@ -83,7 +83,7 @@ class ChannelManagerTest {
             }
             upstream.close(TestException())
             collection.await()
-            fail("collection should propagate upstream exception.")
+            throw AssertionError("collection should propagate upstream exception.")
```
Comment (burakeregar) [inline multicast/src/test/kotlin/com/dropbox/flow/multicast/ChannelManagerTest.kt]: We can actually keep it as is by just importing `org.junit.Assert.fail`

Comment (eyalgu) [reply multicast/src/test/kotlin/com/dropbox/flow/multicast/ChannelManagerTest.kt]: @yigit - curious, does google/Truth have a philosophical objection to a `fail` function or is it an oversight in v1 public release?

**Comment by eyalgu** [comment]
this pr is contained within #32

File: multicast/src/test/kotlin/com/dropbox/flow/multicast/ChannelManagerTest.kt (hunk 1 above)
Comment (yigit) [reply multicast/src/test/kotlin/com/dropbox/flow/multicast/ChannelManagerTest.kt]: i have no idea, it was just not available so i replaced w/ an assertion error.

**Comment by yigit** [comment]
i can abort this, didn't realize it was done, i just thought it was missed.

**Pull request closed without merging**